### DISCUSSION TITLE
Is `babel-plugin-lodash` compatible with ParcelJs?

### DISCUSSION THREAD

**Discussion by bologer** [parcel-bundler/parcel#4711] [General]
I'm trying to optimize main bundle size and I have duplication of lodash in it. 

It costs about 200kb in total and duplicated twice. I've found `babel-plugin-lodash` which seems to be doing the job, however it doesn't work together with Parcel. 

Is there a way to make it work with Parcel? 

Also, is there a way to determine what third party library is bundling lodash twice? I use, `parcel-plugin-bundle-visualiser`, it is nice to see overall picture, but it only shows that lodash was loaded twice and not the origin (from what third party package it is coming from).

**Comment by mischnic** [comment]
> babel-plugin-lodash

That only works with ES6 modules and Parcel only runs Babel on your source code, not on `node_modules` code.

> but it only shows that lodash was loaded twice and not the origin

But doesn't it say something like `node_modules/lodash/index.js` and `node_modules/someLibrary/node_modules/lodash/index.js`? Then you might be able to adjust your own lodash version to make npm/Yarn deduplicate them.

>  Also, is there a way to determine what third party library is bundling lodash twice

You could build with `--no-minify` and trace the imports yourself in the sourcecode.

With Parcel 2, you could use https://bundle-buddy.com/.

**Reply by bologer** [reply]
I see, thank you for useful links. 

One is loading
```
node_modules/lodash
```

and other is 

```
node_modules/lodash-es
```

Here is a foam tree:

<img width="710" alt="Screen Shot 2020-06-07 at 18 56 25" src="https://user-images.githubusercontent.com/9879933/83974862-a01b9100-a8f0-11ea-90d5-6dcf583918fe.png">

**Reply by bologer** [reply]
Sorry, by the following: 
> Then you might be able to adjust your own lodash version to make npm/Yarn deduplicate them.

You mean [transformation](https://parceljs.org/transforms.html)?

**Reply by mischnic** [reply]
No. I mean:

Let's say you use `lodash` in your application directly and some library uses `lodash-es`.
If you replace `lodash` with `lodash-es` in your application ("manually"), you will only get one lodash instance bundled.

**Reply by bologer** [reply]
I see, I think this wouldn't help me, because I don't use lodash in my project, they are grabbed from third party packages which I have no controlled over.
